assets/src/edit-story/app/user/userProvider.js

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import { useCallback, useState } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { useAPI } from '../api';
+import Context from './context';
+
+function UserProvider({ children }) {
```

Comment:
I'm not sure I see the need for a new provider here. I think this could easily go in the API provider and fetch on load like some of the other requests - and make sure it's in prefetched from DB, so it'll be instant?

Also, not fully sure what other work this provider would do - know about users in general? In that case, it's a "UsersProvider" or something like that IMO. But even if that, I don't see it warrant a separate provider.

Reply:
This is all documented in the original ticket - https://github.com/google/web-stories-wp/issues/6092#issuecomment-770844568

Reply:
We can always keep it simple and only add a new provider if really needed.

Reply:
I have done the work now, seems pointless to remove it now.

Reply:
I really don't think adding the caching to the API provider makes any sense at all. The apiProvider just is a wrapper around apiFetch. I believe caching should be done down stream.